src/tracing/esp32_diagnostic_trace/DiagnosticTracing.cpp

```
@@ -0,0 +1,169 @@
+                                              MurmurHash("Resolver") }; // namespace
+
+bool IsPermitted(const char * str)
+{
+    HashValue hashValue = MurmurHash(str);
+    for (HashValue permitted : gPermitList)
+    {
+        if (permitted == 0)
+        {
+            break;
+        }
+        if (hashValue == permitted)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+void ESP32Diagnostics::LogMessageReceived(MessageReceivedInfo & info) {}
+
+void ESP32Diagnostics::LogMessageSend(MessageSendInfo & info) {}
+
+void ESP32Diagnostics::LogNodeLookup(NodeLookupInfo & info) {}
+
+void ESP32Diagnostics::LogNodeDiscovered(NodeDiscoveredInfo & info) {}
+
+void ESP32Diagnostics::LogNodeDiscoveryFailed(NodeDiscoveryFailedInfo & info) {}
+
+void ESP32Diagnostics::LogMetricEvent(const MetricEvent & event)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    switch (event.ValueType())
+    {
+    case ValueType::kInt32: {
+        ESP_LOGI("mtr", "The value of %s is %ld ", event.key(), event.ValueInt32());
+        Diagnostic<int32_t> metric(event.key(), event.ValueInt32(), esp_log_timestamp());
+        err = mStorageInstance.Store(metric);
+    }
+    break;
+
+    case ValueType::kUInt32: {
+        ESP_LOGI("mtr", "The value of %s is %lu ", event.key(), event.ValueUInt32());
+        Diagnostic<uint32_t> metric(event.key(), event.ValueUInt32(), esp_log_timestamp());
+        err = mStorageInstance.Store(metric);
+    }
+    break;
+
+    case ValueType::kChipErrorCode:
+        ESP_LOGI("mtr", "The value of %s is error with code %lu ", event.key(), event.ValueErrorCode());
+        break;
+
+    case ValueType::kUndefined:
+        ESP_LOGI("mtr", "The value of %s is undefined", event.key());
+        break;
+
+    default:
+        ESP_LOGI("mtr", "The value of %s is of an UNKNOWN TYPE", event.key());
+        break;
+    }
+
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(DeviceLayer, "Failed to store Metric Diagnostic data"));
```

should we propagate the error a level up?

---

Here it is not possible as this method is overridden from backend.